### PR TITLE
Fixes for Currency Field to work as Controlled Component

### LINKED ISSUE #121: Currency field is not working as expected
Describe the bug:
Currency field value is not getting formatted as expected. 

Steps To Reproduce:
Assign an initial value to the Currency field. The value is getting formatted only on onBlur. It should be formatted for initial existing value also.

Expected behavior:
Currency field value should be formatted for initial exiting values.

### PR DESCRIPTION
**Pull request by sidharthmehra**
Fix initial formatting of value for currency field. 
Update the onFocus and onBlur handlers to work across all countries.
Add controlled and uncontrolled example in the styleguide.

Fixes #121

### CODE REVIEW

**Comment by sidharthmehra** [comment]
screen shot 2018-08-29 at 3 10 50 pm

File: packages/core/src/components/CurrencyField/CurrencyField.js [hunk 1]
```diff
@@ -129,12 +136,21 @@ export class CurrencyField extends Component<CurrencyFieldPropTypes, CurrencyFie
...
+
+  onFocus = (value: string) => {
+    // Triggered when the user focuses onto the field for the first time
+    const { countryCode } = this.props
+    const country = companyCodeMapper(countryCode)
+    const delimiter = country.delimiter
+    const currencyValue = value.replace(/\D/g, match => (match === delimiter ? delimiter : ''))
+    this.setState({
+      formattedValue: currencyValue.toString(),
+      displayedValue: currencyValue.toString(),
```
Comment (colshacol) [inline packages/core/src/components/CurrencyField/CurrencyField.js]: I see formattedValue and displayedValue identical across the entire file, is it possible to eliminate one?

Comment (sidharthmehra) [reply packages/core/src/components/CurrencyField/CurrencyField.js]: Good catch. Thought of doing that initially. Let me make the fix.

Comment (sidharthmehra) [reply packages/core/src/components/CurrencyField/CurrencyField.js]: Done.

**Pull request merged**